### PR TITLE
Eq 1114 make question skip conditions consistent with groups

### PR DESCRIPTION
**Pull request by dwyeradam**
### What is the context of this PR?
Allow for Questions to be skipped with an array of possible (or) conditions in line with those used for Groups and Blocks.
Allow for the summary page to check if a question has been skipped by metadata values.\

### How to review

The test_skip_condition_question schema has been created with the new style skip array. On the first page you can select Yes to either/both of the skip questions. If yes is selected on either then the second question will not be present on the next page. You can also set your user_id to "Skip" and that will skip the second question based on metadata.

### CODE REVIEW

File: app/templating/summary/question.py [hunk 1]
```diff
@@ -17,17 +17,15 @@ def __init__(self, question_schema, answers):
...
-                    answer = answer.unescape()
-
-                if not evaluate_rule(when_rule, answer):
-                    return False
-            return True
-
+            for when_rules in self.skip_condition:
+                for when_rule in when_rules['when']:
+                    if 'id' in when_rule:
+                        answer = answer_store.map().get(when_rule['id'])
```
Comment (ajmaddaford) [inline app/templating/summary/question.py]: We shouldn't use the map method to get an answer, we should be using the `filter` method on `answer_store` to return all matching answers.

File: app/templating/summary/question.py [hunk 2]
```diff
@@ -17,17 +17,15 @@ def __init__(self, question_schema, answers):
...
+                        if isinstance(answer, Markup):
+                            answer = answer.unescape()
```
Comment (ajmaddaford) [inline app/templating/summary/question.py]: `answer` isn't used

File: app/templating/summary_context.py [hunk 3]
```diff
@@ -27,6 +27,6 @@ def build_summary_rendering_context(schema_json, answer_store, metadata):
                 if isinstance(value, str):
                     answers[answer_id] = escape(value)
 
-            groups.extend([Group(group, answers, path, metadata, url_for)])
+            groups.extend([Group(group, answer_store, path, metadata, url_for)])
```
Comment (ajmaddaford) [inline app/templating/summary_context.py]: Changing this to use the answer_store rather than answers means answers aren't being escaped.

**Review by ajmaddaford — CHANGES_REQUESTED** [review]

**Comment by ajmaddaford** [comment]
I think it would be good to get together with @adoublebarrel tomorrow to review the changes to ensure we don't lose any of the escaping fixes he recently added.

**Review by ajmaddaford — APPROVED** [review]

File: tests/functional/spec/skip-condition-question.spec.js [hunk 4]
```diff
@@ -0,0 +1,126 @@
+import {getRandomString, openQuestionnaire, startQuestionnaire} from '../helpers'
+
+import DoYouWantToSkip from '../pages/surveys/skip_condition_question/do-you-want-to-skip.page.js'
+import ShouldSkip from '../pages/surveys/skip_condition_question/should-skip.page.js'
+import SummaryPage from '../pages/summary.page'
+
+describe('Skip condition question', function() {
```
Comment (jonnyshaw89) [inline tests/functional/spec/skip-condition-question.spec.js]: shouldn't we be making new tests under `functional_new`

File: data/schema/schema_v1.json [hunk 5]
```diff
@@ -364,7 +364,7 @@
                                                         },
                                                         "skip_condition": {
```
Comment (jonnyshaw89) [inline data/schema/schema_v1.json]: Maybe this should be renamed to `skip_conditions`?

**Review by jonnyshaw89 — DISMISSED** [review]

**Review by jonnyshaw89 — CHANGES_REQUESTED** [review]
See previous comments

**Review by dcdarrell9 — COMMENTED** [review]
Not sure if the tests should be under functional_new like Jon said, but everything else looks good.

**Review by jonnyshaw89 — APPROVED** [review]

**Comment by jonnyshaw89** [comment]
@dwyeradam squash?

**Pull request merged**